css/BEM-style.md

```diff
@@ -0,0 +1,242 @@
+    <h1 class="primary-heading">Want to learn more about BEM?</h1>
+</div>
+```
+
+Like previous example the element for the hero-box would be **.hero-box__text**, that describes the text information associated to the box.
+
+>It is important to note that the heading `<h1></h1>` is defined as it's own block, and this is because it has it's own standalone meaning. Moreover remeber following the principle of DRY, this **primary-heading** can be used again in some other place as and when required.Thus it is named with a class of **.primary-heading** instead of **.hero-box__heading**.
+
+Now let's combine the two elements of hero-box block :'the button' and 'the text' into the hero-box block element. The markup would look like this:
+
+```
+<section class="first-section">
+    <div class="hero-box">
+        <div class="hero-box__text">
+            <h1 class="primary-heading">Want to learn more about BEM?</h1>
+        </div>
+        <div class="hero-box__links">
+            <a href="#"class="link link--primary">Yes, please!</a>
+            <a href="#"class="link link--primary-inverted">No, I'm okay!</a>
+        </div>
+    </div>
+</section>
+```
+
+You can now pretty much style all the components the way you like, just by using the names of the classes only (remember BEM was all about making specificity low and avoiding conflicts).
+>If you are using SaSS , BEM becomes more powerful because now you can use the Sass nesting feature for all the classes.
+```
+.hero-box{
+    height: 100vh;
+    background:linear-gradient(to bottom right,#000,transparent),
+    url("https://images.unsplash.com/photo-1580554996018-ff8b408fc162?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80");
+     background-position: center;
+    background-repeat: no-repeat;
+    background-size: cover;
+    display:flex;
+    flex-direction:column;
+    justify-content: center;
+    align-items: flex-start;
+    
+    &__text{
+       margin:2rem;
+    } 
+    &__links{
+        margin:0.5rem;   
+    }  
+}
+.link{
+    text-decoration: none;
+    border:1px solid saddlebrown;
+    padding:1rem 2rem;
+    color:white;
+    border-radius:113rem;
+    font-size:2rem;
+    text-align: center;
+    display:inline-block;
+    margin:2rem;
+    &--primary{
+         background:saddlebrown;
+         &:hover{
+             background-color:rgb(201, 93, 17);
+         }
+    }
+    &--primary-inverted{
+        background:transparent;
+        &:hover{
+            background-color:rgb(201, 93, 17);
+        }
+    }
+}
+.primary-heading{
+    font-size:6rem;
+}
+```
+
+**3.How BEM is useful on larger codebases because it is scalable and maintainable.:** 
+
+What if I told you to add a new button with some animation on it in the hero-box section? What other information will you need? Well nothing, since all necessary information is already present with you now and you know exactly where this component must lie and how you will name the classes on it. 
```

Comment:
Can you provide an example please?

Reply:
@alodahl I have committed the changes. Hope they're okay now?? You can let me know if more changes are needed. Thank you.